tests/check/check_signed_video_sign.c

```diff
@@ -721,6 +721,81 @@ START_TEST(golden_sei_created)
 }
 END_TEST
 
+/* Test description
+ * Generates SEIs using golden SEI prinsiple and verifies them.
+ * The operation is as follows:
+ * 1. Setup a signed_video_t session
+ * 2. Generate and store golden SEI
+ * 3. End signed_video_t and start a new one
```

Comment:
The operations 1-3 do not add any value to the tests, since it is already checked in golden_sei_created.
The operations 4-7 are independent from 1-3 and is fine to test here, but I think you should add a check that the public key is not part of the SEI.

The test you've written better fits the auth side, where you should verify that the validation is successful with the golden SEI principle.